helpers/aws.js

```diff
@@ -158,30 +158,38 @@ class AwsClient {
     }
 
     /**
-     * Upload command as Stream
-     * @method uploadCmdAsStream
-     * @param {Object}             config                Config object
-     * @param {Buffer}              config.payload       Payload to upload
+     * Upload object directly
+     * @method uploadObject
+     * @param {Object}              config               Config object
+     * @param {Object}              config.payload       Payload to upload
      * @param {String}              config.objectKey       Path to Object
      * @return {Promise}
      */
-    uploadCmdAsStream({ payload, objectKey }) {
-        // stream the data to s3
-        const passThrough = new stream.PassThrough();
+    uploadObject({ payload, objectKey }) {
+        const now = new Date();
+        const metadata = {
+            stored: now.toString()
+        };
+        let type = 'application/octet-stream';
+        let prettyPayload;
+
+        try {
+            prettyPayload = JSON.stringify(payload);
```

Comment:
why do we have to convert to json ? isn't payload just string ?

Reply:
It is an Object
const contents = {
                    c: request.payload,
                    h: {}
}

Reply:
But why do we have to pass it as an object https://github.com/screwdriver-cd/store/pull/113/files#diff-ebcc8b02c1b6015c4779f8a1febc7d667cf834cbb9640ba863ceed49ed8f8cc8L104-L107 ?